ssl/statem/statem_clnt.c

```diff
@@ -1362,7 +1364,22 @@ MSG_PROCESS_RETURN dtls_process_hello_verify(SSL_CONNECTION *s, PACKET *pkt)
         return MSG_PROCESS_ERROR;
     }
     s->d1->cookie_len = cookie_len;
+    if (ssl == NULL) {
+        SSLfatal(s, SSL_AD_DECODE_ERROR, SSL_R_LENGTH_MISMATCH);
+        return MSG_PROCESS_ERROR;
+    }
+
+    max_version = SSL_get_max_proto_version(ssl);
 
+    /*
+     * Server responds with a HelloVerify which means we cannot negotiate a
+     * higher version than DTLSv1.2.
+     */
+    if ((max_version == DTLS1_3_VERSION || max_version == 0)
+            && !SSL_set_max_proto_version(ssl, DTLS1_2_VERSION)) {
+        SSLfatal(s, SSL_AD_DECODE_ERROR, SSL_R_LENGTH_MISMATCH);
```

Comment:
Again the AD and R values need to be changed IMO. I.e. the SSL_set_max_proto_version would fail if we had DTLS1.3 as the min version -> probably some protocol version mismatch should be reported instead.

Reply:
Agree. Too fast with the copy-paste.